joomla/joomla-cms
# [4] Fix trash menu items displaying logic

### Summary of Changes

Another bug found while developing Empty States :) One a security bug.

These bugs are in  Languages, Newsfeeds, Tags and Notes. 

The bug is that the code below makes no sense. What its TRYING to do is only show the Trash button (in the Actions dropdown menu) when the filter for the published dropdown is NOT trashed. 

But what it ACTUALLY is doing (by mistake) is making it IMPOSSIBLE to Trash items IF you have a filter set (Select Published from the Status filter for example)

```php
if (!$this->state->get('filter.published') == -2)
```

An example test case showing the difference in the code is here https://3v4l.org/XTPkK

That is resolved in this PR


### Testing Instructions


 - Login to Joomla 4 Admin
 - Navigate to Components Tags
 - Create a Tag and set it published
 - Select the tag from the list of tags 
 - Click Actions - NOTE THERE IS A TRASH BUTTON (Dont click it) 
 - Click Filter Options
 - Select PUBLISHED in the STATUS dropdown filter
 - Select the tag from the list of tags 
 - Click Actions - NOTE THERE IS A NOT TRASH BUTTON (This is the bug, you should be able to trash this tag here)
 - Apply PR
 - Click Actions - NOTE THERE IS NOW A TRASH BUTTON (This is the fixed bug)


### Actual result BEFORE applying this Pull Request

Component | Trash Button Bug
------------ | ------------- 
Languages |  **Has Trash Button Bug**
Newsfeeds |  **Has Trash Button Bug**
Tags |  **Has Trash Button Bug**
User Notes | **Has Trash Button Bug**

 
### Expected result AFTER applying this Pull Request

Component |  Trash Button Bug
------------ | -------------
ALL * | No Trash Button Bug

### Documentation Changes Required

None
